docs/advanced-features/middleware.md

```diff
@@ -34,7 +34,7 @@ npm install next@latest
 import type { NextFetchEvent, NextRequest } from 'next/server'
 
 export function middleware(req: NextRequest, ev: NextFetchEvent) {
-  return new Response('Hello, world!')
+  return new Response(null, { headers: { location: '/hello-world' } })
```

Comment:
This seems fairly weird as an example of what to do, especially as it is just as much breaking as returning the response body because it can't be handled correctly when client-side routing.

Reply:
`new Response` shouldn't be used in any case, the only allowed usage is:
- `NextResponse.redirect()`
- `NextResponse.rewrite()`
- `NextResponse.next()`
- Returning `undefined` which is the same as `NextResponse.next()`

Reply:
Good call, let me update this.

Reply:
For the record, `NextResponse.next()` does not return `undefined`, but a response with a null ([default](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response#parameters)) body.